bazel/utils/clang_format.bzl

```diff
@@ -0,0 +1,48 @@
+def _clang_format(ctx):
+    clang_format_file = ctx.file.clang_format
+    out_file = ctx.actions.declare_file(ctx.label.name + ".bash")
+    substitutions = {
+        "{pattern-to-format}": ctx.attr.pattern_to_format,
+        "{clang-format}": clang_format_file.short_path,
+        "{style}": ctx.attr.style,
+    }
+    ctx.actions.expand_template(
+        template = ctx.file._template,
+        output = out_file,
+        substitutions = substitutions,
+        is_executable = True,
+    )
+
+    return DefaultInfo(
+        files = depset([out_file]),
+        runfiles = ctx.runfiles(files = [clang_format_file]),
+        executable = out_file,
+    )
+
+clang_format = rule(
+    implementation = _clang_format,
+    attrs = {
+        "style": attr.string(
+            mandatory = True,
+            # From clang-format-7 --help output.
+            values = ["LLVM", "Google", "Chromium", "Mozilla", "WebKit", "file"],
+            doc = "The coding style to apply.",
+        ),
+        # TODO: Support a list of patterns to format.
+        "pattern_to_format": attr.string(
```

Comment:
I would just call this "pattern" (or patterns). How do we do ".cc", ".h", ".cpp", ".hpp", ".c" file if only one pattern is supported? Well, I guess we can write a more advanced pattern for find? Is that what you had in mind?

Reply:
So, as of now the solution is to simply have multiple `clang_format` rules instantiations. Kinda ugly, but I'm not exactly sure about all our possible use cases. Anyway, yes, the simplest solution would be to pass a regex to the `find` command (this requires a small change in the bash script).